### PR TITLE
Get build counts from Travis for public org.

### PR DESCRIPTION
**Pull request by benpatterson**
This script will interrogate the Travis REST API to
obtain information on any running builds for the active
repos. (Note: it will only get the first 100 active repos.)

Output looks like this snipped example when run with
debug logging enabled:

```bash
$ python -m travis.build_info --org edx --log-level debug
2016-03-04 15:41:42,007 [DEBUG] --->edx-analytics-dashboard
2016-03-04 15:41:42,008 [DEBUG] total: 1, started: 0
2016-03-04 15:41:42,346 [DEBUG] --->edx-documentation
2016-03-04 15:41:42,346 [DEBUG] total: 0, started: 0
2016-03-04 15:41:42,711 [DEBUG] --->edx-ora2
2016-03-04 15:41:42,711 [DEBUG] total: 0, started: 0
2016-03-04 15:41:43,771 [DEBUG] --->configuration
2016-03-04 15:41:43,771 [DEBUG] total: 2, started: 0
2016-03-04 15:41:53,969 [DEBUG] --------
2016-03-04 15:41:53,969 [INFO] overall_total=3
2016-03-04 15:41:53,969 [INFO] overall_started=0
2016-03-04 15:41:53,969 [INFO] overall_queued=3
```

### CODE REVIEW

**Comment by benpatterson** [comment]
@jzoldak

File: travis/build_info.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+logger = logging.getLogger(__name__)
+
+BASE_URL = 'https://api.travis-ci.org/'
+
+
+def get_repos(org):
+    """
+    Returns list of active repos in a given org.
+    """
+
```
Comment (jzoldak) [inline travis/build_info.py]: I realize it's a style thing, so whatever you decide is totally cool. I'd suggest though that we not have a blank line under docstrings. IMO it's not needed for readability and makes the files longer.

File: travis/build_info.py [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    repo_list = []
+    # Note: One call is needed if org has <= 100 active repos.
+    # If org has > 100, this code will need to be modified
+    #  to include 'offset' requests
```
Comment (jzoldak) [inline travis/build_info.py]: Are we close to 100? How will we know when we need to implement this?

File: travis/build_info.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    req = requests.get(
+        BASE_URL + 'v3/owner/{org}/repos?active=true'.format(org=org)
+    )
+    if req.status_code != 200:
+        raise requests.HTTPError(req.status_code)
+    req_json = req.json()
+    try:
+        for i in req_json['repositories']:
```
Comment (jzoldak) [inline travis/build_info.py]: I was surprised that i as a variable name got past the linter. Then checked and realized that we're not linting this repo. :disappointed:

File: travis/build_info.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+    """
+
+    repo_slug = '{org}/{repo}'.format(org=org, repo=repo)
+    req = requests.get(
+        BASE_URL + 'repos/{repo_slug}/builds'.format(repo_slug=repo_slug)
+    )
+    build_list = req.json()
+    active_build_list = []
+    for build in build_list:
+        if build['state'] != 'finished':
```
Comment (jzoldak) [inline travis/build_info.py]: I recommend that we either put statements that refer directly to keys in try blocks and catch KeyErrors or use the get() method for dicts, in this method and others. E.g. build.get('state')

File: travis/build_info.py [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+            active_build_list.append(build)
+    return active_build_list
+
+
+# @property
```
Comment (jzoldak) [inline travis/build_info.py]: leftovers?

File: travis/build_info.py [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+            started_count += 1
+
+    return build_count, started_count
+
+
+def repo_active_job_count(active_builds):
+    """
+    Returns count of active jobs associated with builds in a
+    given repo.
+    """
```
Comment (jzoldak) [inline travis/build_info.py]: Can you please document the possible values for build stage and job state, or add a URL to the API specs? 
I couldn't find what the stage values might be with a quick google search. 
For state values, it seem from [states.rb#L6](https://github.com/travis-ci/travis.rb/blob/dcc9f20535c811068c4ff9788ae9bd026a116351/lib/travis/client/states.rb#L6) that the values are: created queued received started passed failed errored canceled ready
For build stages I don't see those [in the API spec](https://docs.travis-ci.com/api#builds), maybe I'm looking in the wrong place?

File: travis/build_info.py [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+    )
+    parser = argparse.ArgumentParser(description=desc)
+    parser.add_argument(
+        '--org', '-o',
+        dest='org',
+        help='Travis org',
+        required=True
+        )
+    parser.add_argument(
+        '--task-class',  # this is not doing anything for now.
```
Comment (jzoldak) [inline travis/build_info.py]: What is this for? If it's not being used, I would leave it out until it is needed.

File: travis/build_info.py [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+        default="INFO",
+    )
+    args = parser.parse_args(raw_args)
+
+    # Set logging level
+    logging.getLogger(__name__).setLevel(args.log_level.upper())
+    get_build_counts(org=args.org)
+
+
+def get_build_counts(org):
```
Comment (jzoldak) [inline travis/build_info.py]: By convention, I would put this method up above the main method in this file.

File: travis/build_info.py [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+    """
+    actual script execution
```
Comment (jzoldak) [inline travis/build_info.py]: This could use a better docstring.

File: travis/build_info.py [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+    """
+
+    repos = get_repos(org)
+    main_count = 0
```
Comment (jzoldak) [inline travis/build_info.py]: Rename this to org_build_count?

File: travis/build_info.py [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+    main_started_count = 0
```
Comment (jzoldak) [inline travis/build_info.py]: Rename this to org_started_count?

File: travis/build_info.py (hunk 2 above)
Comment (benpatterson) [inline travis/build_info.py]: We are currently at 67 that are configured for Travis usage. However, the number that actually runs builds is below that (maybe 1 or 2 dozen). From what I can tell, the Travis API lists according to the recent usage. But obviously an off-hand observation is not a solution. I'll put in a TE ticket to address this.

File: travis/build_info.py (hunk 5 above)
Comment (benpatterson) [inline travis/build_info.py]: Yeah that's a to-do. Subsequent pull request will follow this one.

File: travis/build_info.py (hunk 7 above)
Comment (benpatterson) [inline travis/build_info.py]: Yeah still a to-do. Honestly I'm close to the other pull request and would rather leave this in for now. It is a no-op in this commit anyway.

File: travis/build_info.py (hunk 2 above)
Comment (benpatterson) [inline travis/build_info.py]: https://openedx.atlassian.net/browse/TE-1208

File: travis/build_info.py (hunk 4 above)
Comment (benpatterson) [inline travis/build_info.py]: Done. (Using build.get in most cases, with a try/except in 1 case)

File: travis/build_info.py (hunk 1 above)
Comment (benpatterson) [inline travis/build_info.py]: Done

**Pull request merged**